src/Demos/HSP.lagda

```diff
@@ -268,65 +222,53 @@ f ⟨∘⟩ g = record  { f = (_⟨$⟩_ f) ∘ (_⟨$⟩_ g)
 \end{code}
 \fi
 
-\paragraph*{Inverses of setoid functions}
-We begin by defining an inductive type that represents the \emph{image} of a function.
-%\footnote{cf.~the \ualmodule{Overture.Setoid.Inverses} module of the \agdaalgebras library.}
+\paragraph*{Inverses}
+We frequently need to deal with inverses. These are most easily defined via the
+\emph{image} of a function.
 
 \begin{code}
-
 module _ {𝑨 : Setoid α ρᵃ}{𝑩 : Setoid β ρᵇ} where
  open Setoid 𝑩 using ( _≈_ ; sym ) renaming ( Carrier to B )
 
  data Image_∋_ (f : 𝑨 ⟶ 𝑩) : B → Type (α ⊔ β ⊔ ρᵇ) where
   eq : {b : B} → ∀ a → b ≈ f ⟨$⟩ a → Image f ∋ b
-
 \end{code}
-An inhabitant of \aod{Image} \ab f \aod{∋} \ab b is a dependent pair \AgdaPair{a}{p},
-where \AgdaTyped{a}{A} and \ab p~\as :~\ab b \af{≈} \ab f~\ab a is a proof that
-\ab f maps \ab a to \ab b.  Since the proof that \ab b
-belongs to the image of \ab f is always accompanied by a witness \AgdaTyped{a}{A}, we can
-actually \emph{compute} a range-restricted right-inverse of \ab f, as follows.
-%. For convenience, we define this
-%inverse function and give it the name \af{Inv}.
 
-\begin{code}
+An inhabitant of \aod{Image} \ab f \aod{∋} \ab b is a point \ab
+where \AgdaTyped{a}{A} and a proof \ab p~\as :~\ab b \af{≈} \ab f~\ab a that
+\ab f maps \ab a to \ab b.
+Since the witness that \ab b
+belongs to the image of \ab f is always accompanied by a concrete witness \AgdaTyped{a}{A}, we can
+\emph{compute} a range-restricted right-inverse of \ab f.  For extra certainty, we also verify
+that your witness really is an inverse.
 
+\begin{code}
  Inv : (f : 𝑨 ⟶ 𝑩){b : B} → Image f ∋ b → Carrier 𝑨
  Inv _ (eq a _) = a
 
-\end{code}
-For each \ab b : \afld{B}, given a pair \AgdaPair{a}{p}~\as :~\aod{Image}~\ab f~\aod{∋}~\ab b witnessing the fact that \ab b belongs to the image of \ab f, the function \af{Inv} simply returns the witness \ab a, which is a preimage of \ab b under \ab f.
-Let's formally verify that \af{Inv} \ab f is indeed the (range-restricted) right-inverse of \ab f.
-
-
-\begin{code}
-
  InvIsInverseʳ : {f : 𝑨 ⟶ 𝑩}{b : B}(q : Image f ∋ b) → f ⟨$⟩ (Inv f q) ≈ b
  InvIsInverseʳ (eq _ p) = sym p
 \end{code}
 
-
 \paragraph*{Injective and surjective setoid functions}
 If \ab{f} % : \ab{𝑨} \aor{⟶} \ab{𝑩}
-is a setoid function from % \ab{𝑨} =
-(\ab A, \af{≈ᴬ}) to
-% \ab{𝑩} =
-(\ab B, \af{≈ᴮ}), then we call \ab f \defn{injective} provided
+then we call \ab f \defn{injective} provided
 \as{∀} (\ab{a₀} \ab{a₁} \as : \ab{A}), \ab{f} \aofld{⟨\$⟩} \ab{a₀} \af{≈ᴮ} \ab{f} \aofld{⟨\$⟩} \ab{a₁}
 implies \ab{a₀} \af{≈ᴬ} \ab{a₁}; we call \ab{f} \defn{surjective} provided
 \as{∀} (\AgdaTyped{b}{B}), \as{∃}~(\AgdaTyped{a}{A}) such that \ab{f} \aofld{⟨\$⟩} \ab{a} \af{≈ᴮ} \ab{b}.
+\ifshort
+We omit the straightforward \agda definitions.
+\else
+
 The \agdastdlib represents injective functions on bare types by the
```

Comment:
I think I meant to use the \agdaalgebras macro (instead of \agdastdlib) here...?   Anyway, this only affects the unabridged version, so not important right now.